From 670871b5162176abeebfb94a8c99d017f82fce8d Mon Sep 17 00:00:00 2001
From: Nvertigo <nvertigo67@gmail.com>
Date: Wed, 13 Jun 2018 11:22:08 +0200
Subject: [PATCH] Allow to opt in installing Gallery2 as non-privileged apk.

  Making Gallery2 privileged breaks "Settings->User->(Crop|Take) Picture",
  because priv_app is denied to open system_data_file by a neverallow
  rule from LineageOS/android_system_sepolicy.

  For non privileged apps there is a dedicated rule
  https://github.com/LineageOS/android_system_sepolicy/blob/lineage-15.1/private/untrusted_app_all.te#L47
  to allow read and write system app data files passed over by Binder.

  The reason to make Gallery2 privileged has been issues deleting photos
  on SD card. This has been a hack from early cm-14.1 and has been
  reverted and rereverted. The hack for the hack on cm-14.1 was allowing
  priv-app to read/write system_data_file, what is impossible on
  lineage-15.1 due to an unresolvable neverallow conflict.

  It doesn't make sense to break "Settings->User->(Crop|Take) Picture" in
  order to fix sdcard access on devices WITHOUT sdcard slot.

  If you want to opt in installing Gallery2 as non-privileged apk, set

    DEVICE_HAS_UNPRIVILEGED_GALLERY := true

  in your device tree.

Change-Id: I4d4b07dc3b5c0dea22f032b1a4a93177af3d468b
---
 Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Android.mk b/Android.mk
index 723ed7eeb..24af71d9b 100644
--- a/Android.mk
+++ b/Android.mk
@@ -39,7 +39,9 @@ LOCAL_STATIC_JAVA_AAR_LIBRARIES := \
 
 LOCAL_PACKAGE_NAME := Gallery2
 
+ifeq ($(filter true, $(DEVICE_HAS_UNPRIVILEGED_GALLERY)),)
 LOCAL_PRIVILEGED_MODULE := true
+endif
 
 LOCAL_OVERRIDES_PACKAGES := Gallery Gallery3D GalleryNew3D
 
-- 
2.17.1

